XenAdmin/Dialogs/GraphDetailsDialog.cs

```diff
@@ -208,7 +208,7 @@ private void EnableDataSource()
 
         private void getDataSourcesAction_Completed(ActionBase sender)
         {
-            if (!(sender is GetDataSourcesAction action))
+            if (!(sender is GetDataSourcesAction action) || !action.Succeeded)

```

Comment:
What happens on the UI if the action fails?

Reply:
It shouldn't happen because I think all users can access the list of data sources, but it keeps loading indefinitely.

We should handle this better. We can add some error state.

![image](https://user-images.githubusercontent.com/32554698/235181679-b650d8ef-fcde-4d50-8112-4bab3d366880.png)

Reply:
Hmm, I suspected so because the spinner at least is not hidden elsewhere.
Simplest solution: revert the Succeeded check in both files. We're not accessing `Result` there anyway, so no need to change it.